pkg/apis/pipeline/v1alpha1/task_conversion.go

```diff
@@ -0,0 +1,220 @@
+/*
+Copyright 2019 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha2"
+	"knative.dev/pkg/apis"
+)
+
+func (source *Task) ConvertUp(ctx context.Context, obj apis.Convertible) error {
```

Comment:
This allows us to take a `v1alpha1` object and "convert it up" to a more recent version (that will be passed, by the webhook or the reconcilier for in-memory upgrade)